scripts/joinmarket-qt.py

```diff
@@ -1464,71 +1464,84 @@ def openQRCodePopup(self, address):
         popup.show()
 
     def updateWalletInfo(self, walletinfo=None):
-        nm = jm_single().config.getint("GUI", "max_mix_depth")
-        l = self.walletTree
+        max_mix_depth_count = jm_single().config.getint("GUI", "max_mix_depth")
 
+        previous_expand_states = []
         # before deleting, note whether items were expanded
-        esrs = []
-        for i in range(l.topLevelItemCount()):
-            tli = l.invisibleRootItem().child(i)
+        for i in range(self.walletTree.topLevelItemCount()):
+            tli = self.walletTree.invisibleRootItem().child(i)
             # must check top and also the two subitems (branches):
             expandedness = tuple(
                 x.isExpanded() for x in [tli, tli.child(0), tli.child(1)])
-            esrs.append(expandedness)
-        l.clear()
-        if walletinfo:
-            rows, mbalances, xpubs, total_bal = walletinfo
-            if jm_single().config.get("BLOCKCHAIN", "blockchain_source") == "regtest":
-                self.wallet_name = mainWindow.testwalletname
+            previous_expand_states.append(expandedness)
+
+        self.walletTree.clear()
+
+        # Skip the remaining of this method if wallet info doesn't exist
+        if walletinfo == None:
+            return
```

Comment:
This is needed to get `previous_expand_states` working properly.

Without this `return` here, in the original code logic, it would create tree item children during the first pass even though there was no walletinfo. And when the walletinfo was passed in first time, `previous_expand_states` would incorrectly think there were some expansion states to be restored, causing all items to stay collapsed, rather than expanding some of them base on the conditions.